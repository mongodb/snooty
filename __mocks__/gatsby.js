const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  withPrefix: jest.fn().mockImplementation(str => str),
  useStaticQuery: jest.fn(),
};
