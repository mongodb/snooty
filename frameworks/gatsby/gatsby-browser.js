import React from 'react';
import { ThemeProvider } from '@emotion/react';
import { theme } from './src/theme/docsTheme';
import './src/styles/mongodb-docs.css';
import './src/styles/icons.css';
import './src/styles/global-dark-mode.css';

console.log("gatsby-browser loaded");

export const wrapRootElement = ({ element }) => {
  console.log('wrapRootElement here')
  return <ThemeProvider theme={theme}>{element}</ThemeProvider>};

export const shouldUpdateScroll = ({ routerProps: { location } }) => {
  const { hash, state } = location;
  if (hash) {
    return decodeURI(hash.slice(1));
  }
  if (state?.preserveScroll) {
    return false;
  }
  return true;
};
