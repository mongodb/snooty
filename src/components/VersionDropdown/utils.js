import { generatePathPrefix } from '../../utils/generate-path-prefix';

/**
 * Generates the prefix to be used for a version's URL. The prefix will typically consist of the docs repo's
 * set prefix, with the new version appended at the end. Different environments may return different
 * prefixes because of what's passed into the frontend.
 * @param {string} version The version to include at the end of the prefix.
 * @param {Object} siteMetadata Metadata about the site, including its pathPrefix, project, and snootyEnv.
 * @param {string} siteBasePrefix The current docs site's base prefix to append the version to.
 */
export const generatePrefix = (version, siteMetadata, siteBasePrefix) => {
  const { pathPrefix, snootyEnv } = siteMetadata;

  // pathPrefix is typically passed in for deploy jobs. For development purposes,
  // the returned prefix should follow the same pattern (even though the resulting URL might 404).
  if (!!pathPrefix || snootyEnv === 'development') {
    return `/${siteBasePrefix}/${version}`;
  }

  // For staging, replace current version in dynamically generated path prefix
  return generatePathPrefix({ ...siteMetadata, parserBranch: version });
};
