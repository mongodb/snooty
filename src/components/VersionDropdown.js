import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { useSiteMetadata } from '../hooks/use-site-metadata';
import { generatePathPrefix } from '../utils/generate-path-prefix';
import { normalizePath } from '../utils/normalize-path';
import dropdownStyles from '../styles/version-dropdown.module.css';
import Button from '@leafygreen-ui/button';

const zip = (a, b) => {
  // Zip arrays a and b into an object where a is used for keys and b for values
  const dict = {};
  a.forEach((key, i) => (dict[key] = b[i]));
  return dict;
};

const VersionDropdown = ({
  publishedBranches: {
    version: { published, active },
    git: {
      branches: { published: gitBranches },
    },
  },
  slug,
}) => {
  const siteMetadata = useSiteMetadata();
  const { parserBranch, pathPrefix, project, snootyEnv } = siteMetadata;
  const [hidden, setHidden] = useState(true);

  const prefixVersion = version => {
    // Display as "Version X" on menu if numeric version
    const isNumeric = (version = '') => {
      const [firstWord] = version.split();
      return !isNaN(firstWord);
    };
    return `${isNumeric(version) ? 'Version ' : ''}${version}`;
  };

  const useOutsideHandler = ref => {
    // Close dropdown if user clicks outside of the Version button
    const handleClickOutside = event => {
      if (ref.current && !ref.current.contains(event.target)) {
        setHidden(true);
      }
    };

    useEffect(() => {
      // Bind the event listener
      document.addEventListener('mousedown', handleClickOutside);
      return () => {
        // Unbind the event listener on clean up
        document.removeEventListener('mousedown', handleClickOutside);
      };
    });
  };

  // Zip two sections of data to map git branches to their "pretty" names
  const gitNamedMapping = zip(gitBranches, published);
  const currentBranch = gitNamedMapping[parserBranch] || parserBranch;

  const wrapperRef = useRef(null);
  useOutsideHandler(wrapperRef);

  // Handle cases where EOL versions exist
  const legacyDocsURL = `https://docs.mongodb.com/legacy/?site=${process.env.GATSBY_SITE}`;
  let legacyDocsHTML = '';

  if (published.length > active.length) {
    legacyDocsHTML = (
      <li className="">
        <a className="version-selector" href={legacyDocsURL}>
          Legacy Docs
        </a>
      </li>
    );
  }

  // Don't render dropdown if there is only 1 version of the repo
  if (!active || active.length <= 1) {
    return null;
  }

  const generatePrefix = version => {
    // Manual is a special case because it does not use a path prefix (found at root of docs.mongodb.com)
    const isManualProduction = project === 'manual' && snootyEnv === 'production';
    if (isManualProduction) {
      return `/${version}`;
    }

    // For production builds, append version after project name
    if (pathPrefix) {
      const [project] = pathPrefix.split('/');
      return `/${project}/${version}`;
    }

    // For staging, replace current version in dynamically generated path prefix
    return generatePathPrefix({ ...siteMetadata, parserBranch: version });
  };

  return (
    <div ref={wrapperRef} className="btn-group version-sidebar">
      <Button
        variant="default"
        className={['version-button', 'dropdown-toggle', dropdownStyles.button].join(' ')}
        title="Select version"
        onClick={() => setHidden(!hidden)}
        size="large"
      >
        {prefixVersion(currentBranch)}
        <span className={['caret', dropdownStyles.caret].join(' ')}></span>
      </Button>
      {!hidden && (
        <ul className={['dropdown-menu', dropdownStyles.menu].join(' ')} role="menu">
          {active.map(version => {
            const url = normalizePath(`${generatePrefix(version)}/${slug}`);
            return (
              <li className={currentBranch === version ? 'active' : ''} key={version}>
                <a className="version-selector" href={url}>
                  {prefixVersion(version)}
                </a>
              </li>
            );
          })}
          {legacyDocsHTML}
        </ul>
      )}
    </div>
  );
};

VersionDropdown.propTypes = {
  publishedBranches: PropTypes.shape({
    version: PropTypes.shape({
      published: PropTypes.arrayOf(PropTypes.string).isRequired,
    }).isRequired,
    git: PropTypes.shape({
      branches: PropTypes.shape({
        published: PropTypes.arrayOf(PropTypes.string).isRequired,
        active: PropTypes.arrayOf(PropTypes.string).isRequired,
      }).isRequired,
    }).isRequired,
  }).isRequired,
  slug: PropTypes.string.isRequired,
};

export default VersionDropdown;
