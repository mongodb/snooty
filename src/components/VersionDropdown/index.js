import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { cx, css as LeafyCSS } from '@leafygreen-ui/emotion';
import { uiColors } from '@leafygreen-ui/palette';
import { Option, OptionGroup, Select, Size } from '@leafygreen-ui/select';
import { navigate as reachNavigate } from '@reach/router';
import { generatePrefix } from './utils';
import { useSiteMetadata } from '../../hooks/use-site-metadata';
import { theme } from '../../theme/docsTheme';
import { normalizePath } from '../../utils/normalize-path';
import { assertTrailingSlash } from '../../utils/assert-trailing-slash';
import { baseUrl } from '../../utils/base-url';

const StyledSelect = styled(Select)`
  margin: ${theme.size.small} ${theme.size.medium} ${theme.size.small} ${theme.size.medium};

  & > button {
    background-color: ${uiColors.white};
  }

  span {
    font-size: ${theme.fontSize.default};
  }
  ${'' /* Render version dropdown text in front of the Sidebar text */}
  button {
    z-index: 2;
  }
`;

// Returns true if there are any inactive (EOL'd/'legacy') branches
const needsLegacyDropdown = (branches = []) => {
  const isLegacy = (branch = {}) => !branch['active'];
  return branches.some(isLegacy);
};

// Gets UI labels for supplied active branch names
const getUILabel = (branch) => {
  if (!branch['active']) {
    console.warn(
      `Retrieving branch UI label for legacy/EOL'd/inactive branch: ${branch['gitBranchName']}. This should probably not be happening.`
    );
  }
  return branch['versionSelectorLabel'] || createVersionLabel(branch['urlSlug'], branch['gitBranchName']);
};

// If a UI label is not specified for raw versions (e.g. v1.0), create one
// Example: [1.0 or v1.0 or android-v1.0] -> Version 1.0
const createVersionLabel = (urlSlug = '', gitBranchName = '') => {
  if (!urlSlug && !gitBranchName) {
    console.warn('Unable to create version label - neither gitBranchName nor urlSlug defined');
    return 'Version Name Unknown';
  }

  const label = urlSlug || gitBranchName;
  // If the label is numeric (e.g. "2.0" or "v2.0"), we display "Version 2.0"
  if (!isNaN(label)) {
    return `Version ${label}`;
  } else if (label.startsWith('v') && !isNaN(label.slice(1))) {
    return `Version ${label.slice(1)}`;
  }

  return label;
};

// Returns all branches that are neither in 'groups' nor inactive
const getActiveUngroupedBranches = (branches = [], groups = []) => {
  const groupedBranchNames = groups.map((g) => g['includedBranches']).flat() || [];
  return branches.filter((b) => !groupedBranchNames.includes(b['gitBranchName']) && !!b['active']);
};

// Return a branch object from branches that matches supplied branchName
// Typically used to associate a branchName from 'groups' with a branchName in 'branches'
const getBranch = (branchName = '', branches = []) => {
  const branchCandidates = branches.filter((b) => b['gitBranchName'] === branchName);

  if (branchCandidates.length === 0) {
    console.warn(`Could not find branch in 'branches' with gitBranchName: ${branchName}. Check 'groups'.`);
    return null;
  }

  if (branchCandidates.length > 1) {
    console.warn(`Too many branches with name ${branchName}.`);
    return null;
  }

  return branchCandidates?.[0] || null;
};

const setOptionSlug = (branch) => {
  return branch['urlSlug'] || branch['gitBranchName'];
};

const createOption = (branch) => {
  const UIlabel = getUILabel(branch);
  const slug = setOptionSlug(branch);
  return (
    <Option key={slug} value={slug}>
      {UIlabel}
    </Option>
  );
};

const VersionDropdown = ({ repoBranches: { branches, groups, siteBasePrefix }, slug, eol }) => {
  const siteMetadata = useSiteMetadata();
  const { parserBranch, project } = siteMetadata;

  // Attempts to reconcile differences between urlSlug and the parserBranch provided to this component
  // Used to ensure that the value of the select is set to the urlSlug if the urlSlug is present and differs from the gitBranchName
  const currentUrlSlug = useMemo(() => {
    for (let branch of branches) {
      if (branch.gitBranchName === parserBranch) {
        return setOptionSlug(branch);
      }
    }
    return parserBranch;
  }, [branches, parserBranch]);

  if ((branches?.length ?? 0) < 2) {
    console.warn('Insufficient branches supplied to VersionDropdown; expected 2 or more');
    return null;
  }

  // For exclusively Realm SDK pages, we show a subset of the versions depending
  // on the current page selection. For example, on the Android SDK page, we only
  // show Android SDK versions in the version dropdown box.
  if (project === 'realm' && currentUrlSlug?.startsWith('sdk/')) {
    const currentSdkGroup = groups.find((g) => currentUrlSlug.startsWith(g?.['sharedSlugPrefix']));
    if (currentSdkGroup) {
      groups = [currentSdkGroup];
      const includedBranches = currentSdkGroup['includedBranches'];
      branches = branches.filter((b) => includedBranches.includes(b['gitBranchName']));
    }
  }

  const getUrl = (optionValue) => {
    if (optionValue === 'legacy') {
      return `${baseUrl()}legacy/?site=${project}`;
    }
    const prefixWithVersion = generatePrefix(optionValue, siteMetadata, siteBasePrefix);
    return assertTrailingSlash(normalizePath(`${prefixWithVersion}/${slug}`));
  };

  // Used exclusively by the LG Select component's onChange function, which receives
  // the 'value' prop from the selected Option component
  const navigate = (optionValue) => {
    const destination = getUrl(optionValue);
    reachNavigate(destination);
  };

  const activeUngroupedBranches = getActiveUngroupedBranches(branches, groups) || [];

  const eolVersionFlipperStyle = LeafyCSS`
    & > button {
      background-color: ${uiColors.gray.light2} !important;
      color: ${uiColors.gray.base} !important;
    }
  `;

  // TODO: Unfortunately, the Select component seems to buck the ConditionalWrapper component
  // It would be nice to either use the ConditionalWrapper to disable the OptionGroup
  // OR have the OptionGroup not take up space when a label is empty-string. For now,
  // ungrouped branches are handled in a separate array.
  return (
    <StyledSelect
      allowDeselect={false}
      className={cx(eol ? eolVersionFlipperStyle : '')}
      aria-labelledby="View a different version of documentation."
      defaultValue="master"
      onChange={navigate}
      placeholder={'Select a version'}
      popoverZIndex={3}
      size={Size.Large}
      value={currentUrlSlug}
      usePortal={false}
      disabled={eol}
    >
      {activeUngroupedBranches?.map((b) => createOption(b))}
      {groups?.map((group) => {
        const { groupLabel, includedBranches: groupedBranchNames = [] } = group;
        return (
          <OptionGroup key={groupLabel} label={groupLabel}>
            <>{groupedBranchNames?.map((bn) => createOption(getBranch(bn, branches)))}</>
          </OptionGroup>
        );
      })}
      {needsLegacyDropdown(branches) && <Option value="legacy">Legacy Docs</Option>}
    </StyledSelect>
  );
};

VersionDropdown.propTypes = {
  repoBranches: PropTypes.shape({
    branches: PropTypes.arrayOf(
      PropTypes.shape({
        gitBranchName: PropTypes.string.isRequired,
        versionSelectorLabel: PropTypes.string,
        urlSlug: PropTypes.string,
        active: PropTypes.bool.isRequired,
      })
    ).isRequired,
    groups: PropTypes.arrayOf(
      PropTypes.shape({
        groupLabel: PropTypes.string,
        groupDisplayURLPrefix: PropTypes.string,
        includedBranches: PropTypes.array,
      })
    ),
    siteBasePrefix: PropTypes.string.isRequired,
  }).isRequired,
  slug: PropTypes.string.isRequired,
  eol: PropTypes.bool.isRequired,
};

export default VersionDropdown;
