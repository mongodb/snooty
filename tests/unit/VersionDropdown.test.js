import React from 'react';
import { shallow } from 'enzyme';
import VersionDropdown from '../../src/components/VersionDropdown';
import * as Gatsby from 'gatsby';

const publishedBranches = {
  version: {
    published: ['2.12', '2.11', '2.10', '2.9', '2.8', '2.7', '2.6', '2.5', '2.4', '2.2'],
    active: ['2.12', '2.11', '2.10', '2.9', '2.8', '2.7', '2.6', '2.5'],
    stable: null,
    upcoming: null,
  },
  git: {
    branches: {
      manual: 'master',
      published: ['master', 'v2.11', 'v2.10', 'v2.9', 'v2.8', 'v2.7', 'v2.6', 'v2.5', 'v2.4', 'v2.2'],
    },
  },
};

const publishedBranchesNoLegacy = {
  version: {
    published: ['2.12', '2.11', '2.10', '2.9', '2.8', '2.7', '2.6', '2.5'],
    active: ['2.12', '2.11', '2.10', '2.9', '2.8', '2.7', '2.6', '2.5'],
    stable: null,
    upcoming: null,
  },
  git: {
    branches: {
      manual: 'master',
      published: ['master', 'v2.11', 'v2.10', 'v2.9', 'v2.8', 'v2.7', 'v2.6', 'v2.5'],
    },
  },
};

const publishedBranchesUnversioned = {
  version: {
    published: ['master'],
    active: ['master'],
    stable: 'master',
    upcoming: 'master',
  },
  git: {
    branches: {
      manual: 'master',
      published: ['master'],
    },
  },
};

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      parserBranch: 'master',
      pathPrefix: '/bi-connector/master',
      project: 'bi-connector',
      snootyBranch: 'DOCSP-7502',
      user: 'docsworker',
    },
  },
}));

// Testing a property with legacy docs
describe('when rendered', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<VersionDropdown slug="installation" publishedBranches={publishedBranches} />);
  });

  it('shows the dropdown menu', () => {
    expect(wrapper.find('StyledSelect')).toHaveLength(1);
  });

  it('shows the "master" list item is active', () => {
    expect(wrapper.find('StyledSelect').prop('value')).toBe('master');
  });

  it('has 9 list elements', () => {
    expect(wrapper.find('Option')).toHaveLength(9);
  });

  it('shows `Legacy Docs` as the last list element', () => {
    expect(wrapper.find('Option').last().prop('value')).toBe('legacy');
  });

  it('shows the proper name for master', () => {
    expect(wrapper.find('Option').first().text()).not.toBe('master');
  });
});

describe('when rendering a property without legacy docs', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<VersionDropdown slug="installation" publishedBranches={publishedBranchesNoLegacy} />);
  });

  it('has 8 list elements', () => {
    expect(wrapper.find('Option').children()).toHaveLength(8);
  });

  it('does not show `Legacy Docs` as the last list element', () => {
    expect(wrapper.find('Option').last().prop('value')).not.toBe('legacy');
  });
});

describe('when rendering an unversioned property', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<VersionDropdown slug="installation" publishedBranches={publishedBranchesUnversioned} />);
  });

  it('does not render', () => {
    expect(wrapper.find('StyledSelect')).toHaveLength(0);
  });
});
