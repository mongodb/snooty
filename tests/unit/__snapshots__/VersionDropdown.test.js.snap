// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionDropdown Component renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 8px 24px 8px 24px;
}

.emotion-0>label+button,
.emotion-0>p+button {
  margin-top: 3px;
}

.emotion-0 button {
  z-index: 2;
  background-color: var(--select-button-bg-color);
  color: var(--select-button-color);
}

.emotion-0 button div:last-child svg {
  color: var(--select-button-carot);
}

.dark-theme .emotion-0 button:hover {
  background-color: var(--gray-dark4);
  color: var(--gray-light3);
  border-color: var(--gray-base);
  box-shadow: var(--gray-dark2) 0px 0px 0px 3px;
}

@media only screen and (max-width: 1024px) {
  .emotion-0 div,
  .emotion-0 span {
    font-size: 13px;
  }
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #F9FBFA;
  border-color: #889397;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 36px;
  font-weight: 400;
  background-color: #FFFFFF;
  width: 100%;
  font-size: 13px;
}

.emotion-1:focus {
  outline: none;
}

.emotion-1[disabled],
.emotion-1:disabled {
  pointer-events: none;
}

.emotion-1:active,
.emotion-1:focus,
.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:hover,
.emotion-1:active {
  color: #001E2B;
  background-color: #FFFFFF;
  box-shadow: 0 0 0 3px #E8EDEB;
}

.emotion-1:focus {
  background-color: #FFFFFF;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-1>*:last-child {
  grid-template-columns: 1fr 16px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1>*:last-child>svg {
  justify-self: right;
  width: 16px;
  height: 16px;
}

.emotion-1>*:last-child>svg {
  color: #3D4F58;
}

.emotion-1>*:last-child {
  padding: 0 4px 0 12px;
}

.emotion-1:focus {
  box-shadow: 0 0 0 3px #0498EC;
  border-color: rgba(255, 255, 255, 0);
}

@media only screen and (max-width: 1024px) {
  .emotion-1 {
    height: 36px;
    font-size: 16px;
  }
}

.emotion-2 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-3 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  pointer-events: none;
  position: relative;
  z-index: 0;
  padding: 0 11px;
  gap: 6px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  gap: 4px;
  overflow: hidden;
}

.emotion-5 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 100%;
}

.emotion-6 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #889397;
  height: 16px;
  width: 16px;
  justify-self: left;
}

<div
    class="emotion-0"
  >
    <button
      aria-controls="select-1-menu"
      aria-describedby="select-1-description"
      aria-disabled="false"
      aria-expanded="false"
      aria-invalid="false"
      aria-labelledby="View a different version of documentation."
      class="emotion-1"
      data-leafygreen-ui="button"
      data-testid="leafygreen-ui-select-menubutton"
      id="select-2"
      role="button"
      type="button"
      value="master"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            master
          </div>
        </div>
        <svg
          alt=""
          aria-hidden="true"
          class="emotion-6"
          height="16"
          role="presentation"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8.67903 10.7962C8.45271 11.0679 8.04729 11.0679 7.82097 10.7962L4.63962 6.97649C4.3213 6.59428 4.5824 6 5.06866 6L11.4313 6C11.9176 6 12.1787 6.59428 11.8604 6.97649L8.67903 10.7962Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
